### ISSUE TITLE
Cannot set "root-path" option when running via gunicorn

### ISSUE THREAD

**Issue by soundstripe** [Kludex/uvicorn#703]
Both FastAPI's docs and uvicorn's own docs mention running uvicorn as a gunicorn subprocess in production, and this seems to be a good idea overall.

However all the documentation regarding setting root_path to run behind a reverse-proxy on a subpath is pointed at running uvicorn directly. There does not seem to be a documented way to set up gunicorn/uvicorn together with an application mounted at a subpath.

https://www.uvicorn.org/#running-with-gunicorn

@Midnighter [subclassed the UvicornWorker](https://github.com/tiangolo/fastapi/issues/1207) to add this option. I haven't tried his exact solution but I think it is worth discussing here.

**Comment by Midnighter** [comment]
FYI a full demo app is [available in this repo](https://github.com/Midnighter/fastapi-mount/tree/root-path). I also see more and more people mentioning that they run uvicorn directly without gunicorn and seem to be successful that way.

**Comment by soundstripe** [comment]
Thanks! I’ve seen the full app too and yes it is a great example. Running with gunicorn is mentioned so matter of factly that I never questioned it. Maybe a short discussion of the trade offs would be good.

**Comment by Midnighter** [comment]
On gitter I saw @tomchristie recommend running uvicorn directly so maybe this should be adjusted in the [deployment documentation](https://www.uvicorn.org/deployment/)?

**Comment by danielporto** [comment]
Just bumped into this. So there's still no conclusion on if this will ever be added? 
thanks

**Comment by florimondmanca** [comment]
@danielporto I think the best bet for now, if you'd like to stick to Gunicorn, is to go for @Midnighter's solution — subclassing `UvicornWorker` and setting `CONFIG_KWARGS` as appropriate.

```python
import uvicorn

app = ...

class Worker(uvicorn.workers.UvicornWorker):
    CONFIG_KWARGS = {
        # ...
    }
```

```bash
gunicorn example:app -w 4 -k example.Worker
```

I _assume_ we could accept a PR addng a small hint about this to the docs, yeah? Changing the sentence here:

> Gunicorn provides a different set of configuration options to Uvicorn, so some options such as --limit-concurrency are not yet supported when running with Gunicorn.

To something along the lines of:

> Gunicorn provides a different set of configurations to Uvicorn, so some options are not supported by the default `UvicornWorker` class. If you'd like to pass custom configuration, subclass the worker and set `CONFIG_KWARGS`, like so: ...

**Comment by euri10** [comment]
fixed in #908
